conf/clusterrunner-slave

```diff
@@ -0,0 +1,99 @@
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides:
+# Required-Start:    $remote_fs $syslog
+# Required-Stop:     $remote_fs $syslog
+# Default-Start:     -
+# Default-Stop:      0 1 6
+# Short-Description: ClusterRunner init script
+# Description:       Enable clusterrunner "slave" service provided by daemon.
+### END INIT INFO
+
+dir="/var/lib/clusterrunner"
+cmd="nohup clusterrunner slave"
+user="jenkins"
+
+name=`basename $0`
+pid_file="/var/run/$name.pid"
+stdout_log="/var/log/$name.log"
+stderr_log="/var/log/$name.err"
+
+get_pid() {
+    cat "$pid_file"
+}
+
+is_running() {
+    [ -f "$pid_file" ] && ps -p `get_pid` > /dev/null 2>&1
+}
+
+case "$1" in
+    start)
+    if is_running; then
+        echo "Already started"
+    else
+        echo "Starting $name"
+        cd "$dir"
+        if [ -z "$user" ]; then
+            sudo $cmd >> "$stdout_log" 2>> "$stderr_log" &
+        else
+            sudo -u "$user" $cmd >> "$stdout_log" 2>> "$stderr_log" &
+        fi
```

Comment:
Forgot to comment on this before.
Currently if slave hostname is not properly configured (missing fqdn), even though CR-slave connects to master, later when master tries to send subjob to slave, it fails and slave goes offline.
Currently Terraform provisioners pretty much guaranty that the hostname will be configured correctly, but that may not be the case with autoscaling.
We need to figure out how can we make sure that slave hostname is configured properly before stating slave service.

Reply:
Maybe this helps, but the services will not start by default. That means that if you want the service to wait till there if a FQDN set, then make sure we don't do anything after installing the yum package. Once the FQDN is defined, then start the service and mark it as "enabled" on startup.